include/swift/SIL/Projection.h

```diff
@@ -411,7 +411,8 @@ class ProjectionPath {
   }
 
   ProjectionPath &operator=(ProjectionPath &&O) {
-    *this = std::move(O);
+    if (this != &O)
+      *this = std::move(O);
     O.Path.clear();
```

Comment:
@dcci No, the implementation is bogus since it will infinitely recur. It was probably intended to move the elements of O to this.